src/components/common/YourCompany/ChangeBusinessType.vue

```diff
@@ -376,6 +376,10 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
     if (this.getEntitySnapshot?.businessInfo?.legalType === CorpTypeCd.BC_CCC) {
       return false
     }
+    // If our feature flag is off, don't allow editing of the business type.
+    if (this.supportedEntityTypes == null || this.supportedEntityTypes?.length === 0) {
+      return false
+    }
```

Comment:
Should look at new FF instead.

Reply:
I see what you're doing here... sec

Reply:
You only load the flag when the component is mounted. Normally we'd check the flag in real-time (same call).

Reply:
can change it later in main

Reply:
This flag isn't on or off. It's a list of entity types that support changing types.

So this logic isn't quite right. You need to check the current entity type.

Reply:
If it's empty, it wont show, because there are no supported types.

Reply:
It's inclusive:

![image](https://github.com/bcgov/business-edit-ui/assets/3484109/37ed855a-a0c0-4ce3-9ff8-35166ff263a1)

If that's empty, it shouldn't allow any of the types

Reply:
We don't even need this PR - I think we can just turn off the feature flag

Reply:
I think you're right. We should clean up this code though, so we can allow some entity type edits.